velox/common/caching/CachedFactory.h

```diff
@@ -65,7 +65,9 @@ class CachedFactory {
   // The function returns a pair. The boolean in the pair indicates whether a
   // cache hit or miss. The Value is the generator output for the key if cache
   // miss, or Value in the cache if cache hit.
-  std::pair<bool, Value> generate(const Key& key);
+  std::pair<bool, Value> generate(
+      const Key& key,
+      const Properties properties = Properties());
```

Comment:
`const Properties* properties = nullptr` here and `typename Properties = void` then you don't need to create dummy structs for cache factories that do not need it.

Reply:
Thank you for this suggestion. Updated.